TeapotTests/MockTests.swift

```diff
@@ -210,25 +210,76 @@ class MockTests: XCTestCase {
 
         mockedTeapot.setExpectedHeaders(expectedHeaders)
 
-        mockedTeapot.get("/get", headerFields: [ "foo" : "bar" ]) { result in
-            switch result {
+        let wrongHeaders = [ "foo" : "bar" ]
+        mockedTeapot.get("/get", headerFields: wrongHeaders) { wrongHeaderResult in
+            switch wrongHeaderResult {
             case .success:
                 XCTFail("Request suceceeded which should not have!")
             case .failure(_, let response, let error):
                 XCTAssertEqual(response.statusCode, 400)
-                XCTAssertEqual(error.type, .incorrectHeaders)
+                XCTAssertEqual(error, TeapotError.incorrectHeaders(expected: expectedHeaders, received: wrongHeaders))
             }
         }
 
         mockedTeapot.clearExpectedHeaders()
 
-        mockedTeapot.get("/get") { result in
-            switch result {
+        mockedTeapot.get("/get") { clearedHeaderResult in
+            switch clearedHeaderResult {
             case .success(let json, _):
                 XCTAssertEqual(json?.dictionary?["key"] as? String, "value")
             case .failure(_, _, let error):
                 XCTFail("Unexpected error: \(error)")
             }
         }
     }
+
+    func testExpectedHeadersAreNotCheckedForEndpointOverrideButAreForMainRequest() {
+        let mockedTeapot = MockTeapot(bundle: Bundle(for: MockTests.self), mockFilename: "get")
+
+        let expectedHeaders = [
+            "foo": "bar",
+            "baz": "foo2",
+        ]
+
+        mockedTeapot.setExpectedHeaders(expectedHeaders)
+
+        // If you don't pass the expected headers to the overridden endpoint but do
+        // pass them to the final endpoint, this should succeed.
+        mockedTeapot.overrideEndPoint("overridden", withFilename: "overridden")
+
+        mockedTeapot.get("/overridden") { firstOverriddenResult in
+            switch firstOverriddenResult {
+            case .success(let json, _):
+                XCTAssertEqual(json?.dictionary?["overridden"] as? String, "value")
+                mockedTeapot.get("/get", headerFields: expectedHeaders) { firstGetResult in
+                    switch firstGetResult {
+                    case .success(let json, _):
+                        XCTAssertEqual(json?.dictionary?["key"] as? String, "value")
+                    case .failure(_, _, let error):
+                        XCTFail("Unexpected error hitting primary endpoint: \(error)")
+                    }
+                }
+            case .failure(_, _, let error):
+                XCTFail("Unexpected error hitting overridden endpoint: \(error)")
+            }
+        }
+
+        // If you don't pass the expected headers to the overridden getter OR to the main get method, then this should fail.
+        mockedTeapot.get("/overridden") { secondOverriddenResult in
```

Comment:
Is it an idea to put this test in a separate test case end just not call `mockedTeapot.overrideEndPoint("overridden", withFilename: "overridden")` there? It's hard to see the exact difference between this method call and the above. 

Is'tn the only difference that the `headerFields` aren't actually used in the second method call?

Reply:
So the idea is that here you still hit the overridden endpoint with the required headers set up above, the overridden endpoiint should still work. However, if you don't pass headers to the final endpoint, then the test should fail. 

I kept it in the same test since it's testing the same setup of `teapot`, but I could break it out if you prefer @marijnschilling